### DISCUSSION TITLE
idp-discovery only redirects if user is already linked to idp

### DISCUSSION THREAD

**Discussion by emmarichardson** [sventorben/keycloak-home-idp-discovery#139] [Q&A]
I thought this was working but now I am realizing that it only redirects to IDP (Google in my implementation) is the user is already linked to Google.  Am I missing a setting somewhere or is this the expected behavior?  I have specified the domain in the Google settings and Google shows as available for every user with the correct domain email but this is not redirecting unless the account has already set up the link.

**Comment by sventorben** [comment]
Can you please check if email addresses are marked as validated? The authenticator will not redirect to unvalidated email addresses.

**Reply by emmarichardson** [reply]
They are. The problem is that I have no way to enter the domain that the plugin should be looking for. Is there somewhere in the code I can just hard code it in? I cannot get connected to the admin cli.

**Reply by sventorben** [reply]
You may want to look at the code here https://github.com/sventorben/keycloak-home-idp-discovery/blob/main/src/main/java/de/sventorben/keycloak/authentication/hidpd/HomeIdpDiscoverer.java#L69

**Comment by emmarichardson** [comment]
If I read that right, I should be able to replace (config.userAttribute ()
, domain) with (mydomain.com)?

**Comment by emmarichardson** [comment]
I finally figured out how to get logged into kcadmn and was able to set the config.  Thanks for bearing with me.  It is now working as I expected.
